dcs/core/data_repository.py

```diff
@@ -191,29 +194,31 @@ async def cleanup_outbox(self):
         threshold = utc_dates.now_as_utc() - timedelta(days=self._config.cache_timeout)
 
         # filter to get all files in outbox that should be removed
-        outbox_ids = await self._object_storage.list_all_object_ids(
+        object_ids = await self._object_storage.list_all_object_ids(
             bucket_id=self._config.outbox_bucket
         )
-        for outbox_id in outbox_ids:
+        for object_id in object_ids:
             try:
-                drs_object = await self._drs_object_dao.get_by_id(outbox_id)
+                drs_object = await self._drs_object_dao.find_one(
+                    mapping={"object_id": object_id}
+                )
             except ResourceNotFoundError as error:
                 raise self.CleanupError(
-                    object_id=outbox_id, from_error=error
+                    object_id=object_id, from_error=error
                 ) from error
 
             # only remove file if last access is later than cache timeout days ago
             if drs_object.last_accessed <= threshold:
                 try:
                     await self._object_storage.delete_object(
-                        bucket_id=self._config.outbox_bucket, object_id=outbox_id
+                        bucket_id=self._config.outbox_bucket, object_id=object_id
                     )
                 except (
                     self._object_storage.ObjectError,
                     self._object_storage.ObjectStorageProtocolError,
                 ) as error:
                     raise self.CleanupError(
-                        object_id=outbox_id, from_error=error
+                        object_id=object_id, from_error=error
                     ) from error
 
     async def register_new_file(self, *, file: models.DrsObject):
```

Comment:
This should consume a creation model that does not include the object ID. The object ID can then be chosen by this method or we modify the DAO to use the object_id as primary identifier and switch to the surrogate ID DAO instead of the natural ID DAO. Then the ID will be automatically set.

Reply:
The input for the functions tends to be the file ID, and the Object ID is a bucket-specific attribute of the thing that is more persistently represented by the file ID. I'm not sure that the creation model approach is the best route here if it means using the object ID as the primary ID.